src/sqlfluff/core/rules/base.py

```diff
@@ -203,6 +207,30 @@ def __eq__(self, other):
             return False
         return True  # pragma: no cover TODO?
 
+    @classmethod
+    def delete(cls, anchor: BaseSegment) -> "LintFix":
+        """Delete supplied anchor segment."""
+        return cls("delete", anchor)
+
+    @classmethod
+    def replace(cls, anchor: BaseSegment, edit: Iterable[BaseSegment]) -> "LintFix":
+        """Replace supplied anchor segment with the edit segments."""
```

Comment:
@barrywhart actually that's not how replace works. Refer here: https://github.com/sqlfluff/sqlfluff/blob/bb7fbd4c2c0d485079853846b1d306549013058e/src/sqlfluff/core/parser/segments/base.py#L938

This is how the "replace" (previously "edit") type works (and has always worked):
Let's for example say at some level of the parse tree with have these sibling segments
`(segment1, segment2, segment3)`
we then have a `LintFix` to replace `segment1` with `segment_a` and `segment_b`.
`LintFix.replace(segment1, [segment_a, segment_b])`
The fix gets applied at the SAME level of the parse tree as the anchor segment!
so you get
`(segment1, segment_a, segment_b, segment3)`
(all lint fixes get applied to the same level as the anchor segment)

You can also see why I call it replace rather than edit.

Reply:
The key point to take away regarding incorrect parse trees from fixes is:
It is the PLACEMENT (i.e. `anchor_segment`) of the fix that causes these issues NOT the type of the fix.

example:
A rule that adds a whitespace after an identifier:
consider a tree like this:
```
table_identifier
|__(object_identifier_a, dot, object_identifier_b)
```
some rule then determines there should be a whitespace after the table identifier but since its maybe using raw segments it uses object_identifier as it's fix anchor and does a create_after (because the issue is not anything specifically about edit segments): `LintFix.create_after(object_identifier_b, [WhitespaceSegment()])`
this would give a tree like:
```
table_identifier
|__(object_identifier_a, dot, object_identifier_b, whitespace_segment)
```
which is obviously wrong, and leads to later rules not working as expected.

The correct approach would be `LintFix.create_after(table_identifier, [WhitespaceSegment()])` to give:
table_identifier
|__(object_identifier_a, dot, object_identifier_b)
whitespace_segment
```
which would be the same parse tree as if you had re-parsed the fixed string

Reply:
So the challenge is around how you determine the correct anchor_segment placement, not what edit type is used

Reply:
also here is not the place to further discuss the above, will create a separate issue for that, but suffice to say that won't make this suggested change 👍